jujugui/static/gui/src/app/components/generic-input/generic-input.js

```diff
@@ -38,7 +39,7 @@ YUI.add('generic-input', function() {
     },
 
     getInitialState: function() {
-      return {errors: null};
+      return {errors: null, focus: false};
```

Comment:
"focus" was to remain consistent with the CSS class that the state toggles on and off. That CSS class is "focus" to be consistent with the `:focus` pseudo selector. In this case, when 'input:focus' is in effect, we want the label element to also have a `focus` class, which we achieve by having the input's onFocus event handler set the `focus` attribute in state.

All that said, I went back and forth between "focus" and "focused" and don't care that strongly. I gave a slight edge to "focus" for the reason above, but would be fine with changing it.